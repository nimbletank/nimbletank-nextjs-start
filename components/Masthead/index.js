import React from 'react'

const Masthead = ({ title, links }) => {
  return (
    <div>
      <h1>{title}</h1>
      {links &&
        links.map((item, index) => {
          return (
            <a href={item.href} key={`masthead${index}`}>
              {item.text}
            </a>
          )
        })}
    </div>
  )
}

export default Masthead
