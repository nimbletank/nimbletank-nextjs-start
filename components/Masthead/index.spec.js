import React from 'react'
import { shallow } from 'enzyme'
import Masthead from './index'
import serializer from 'enzyme-to-json/serializer'
expect.addSnapshotSerializer(serializer)

const links = [
  {
    href: 'http://zander.wtf',
    text: 'Zander'
  },
  {
    href: 'http://nimbletank.com',
    text: 'Nimbletank'
  }
]

test('Masthead snapshot test', () => {
  const component = shallow(<Masthead title='Zander' links={links} />)
  expect(component).toMatchSnapshot()
})

it('should render a Masthead with title', () => {
  const wrapper = shallow(<Masthead title='Events' />)
  expect(wrapper.instance().props.title).toEqual('Events')
})
