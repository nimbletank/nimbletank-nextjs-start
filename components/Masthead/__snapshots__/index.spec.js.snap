// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead snapshot test 1`] = `
<div>
  <h1>
    Zander
  </h1>
  <a
    href="http://zander.wtf"
  >
    Zander
  </a>
  <a
    href="http://nimbletank.com"
  >
    Nimbletank
  </a>
</div>
`;
